src/main/java/de/rwth/idsg/steve/ocpp/ChargePointServiceInvokerImpl.java

```diff
@@ -235,4 +236,14 @@ public void triggerMessage(ChargePointSelect cp, TriggerMessageTask task) {
             chargePointServiceJsonInvoker.runPipeline(cp, task);
         }
     }
+
+    // -------------------------------------------------------------------------
+    // "Improved security for OCPP 1.6-J" additions. Only for JSON
+    // -------------------------------------------------------------------------
+
+    public void extendedTriggerMessage(ChargePointSelect cp, ExtendedTriggerMessageTask task) {
+        if (cp.isJson()) {
```

Comment:
It would be better to throw an exception when the CP isn’t JSON, rather than silently ignoring the case. This would make misconfigurations easier to detect.

Reply:
done in https://github.com/steve-community/steve/pull/1854/commits/d8c089e00c810e8b28638bcfdf4836df4f58cd5d